libnavigator/src/androidTest/java/com/mapbox/navigation/navigator/internal/NavigatorTest.kt

```diff
@@ -0,0 +1,123 @@
+package com.mapbox.navigation.navigator.internal
+
+import androidx.annotation.RawRes
+import androidx.test.espresso.Espresso
+import androidx.test.platform.app.InstrumentationRegistry
+import com.mapbox.api.directions.v5.DirectionsCriteria
+import com.mapbox.api.directions.v5.models.DirectionsResponse
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.api.directions.v5.models.RouteOptions
+import com.mapbox.geojson.Point
+import com.mapbox.navigation.navigator.internal.util.readRawFileText
+import com.mapbox.navigation.navigator.internal.util.runOnMainSync
+import com.mapbox.navigator.CacheFactory
+import com.mapbox.navigator.CacheHandle
+import com.mapbox.navigator.ConfigFactory
+import com.mapbox.navigator.ConfigHandle
+import com.mapbox.navigator.Navigator
+import com.mapbox.navigator.NavigatorConfig
+import com.mapbox.navigator.ProfileApplication
+import com.mapbox.navigator.ProfilePlatform
+import com.mapbox.navigator.Routes
+import com.mapbox.navigator.SettingsProfile
+import com.mapbox.navigator.TilesConfig
+import org.junit.Assert.assertTrue
+import org.junit.Test
+
+class NavigatorTest {
+
+    private companion object {
+        @RawRes
+        private val DIRECTIONS_RESPONSE =
+            com.mapbox.navigation.navigator.test.R.raw.directions_response_two_leg_route
+    }
+
+    @Test(timeout = 10000)
+    fun successfulSetRoutesToNN() {
```

Comment:
why do we test NN router separately?

Reply:
It runs to make sure that our mapping from `DirectionsRoute` to `DirectionsResponse` it working right here and for future regressions

Reply:
Maybe we won't need to test it if we reuse mapping as I suggested in the [comment](https://github.com/mapbox/mapbox-navigation-android/pull/5500#discussion_r816633264).

BTW, isn't the mapping tested by `SanityCoreRouteTest.route_completes`. The test wouldn't pass if the mapping didn't work.

Reply:
nah, there we don't communicate with native Navigator and don't know the response of it. 

This test is about low-level mapping, with it we can guarantee that do mapping right. It should help us catch regression even if NN change the expected param (String) structure

Reply:
`MapboxNavigation` uses `NativeNavigator` under the hood. In case of regression from NN side `SanityCoreRouteTest.route_completes` will fail, won't it?
What kind of regression can `successfulSetRoutesToNN` catch which `SanityCoreRouteTest.route_completes` can't?